python/cuml/tsa/arima.pyx

```diff
@@ -335,9 +337,15 @@ class ARIMA(Base):
             raise ValueError("ERROR: Invalid order. "
                              "Required: max(p+s*P, q+s*Q) <= 1024")
 
+        original_dtype = determine_array_dtype(endog)
+        if original_dtype != np.float64:
+            warnings.warn("Only float64 is currently supported. `endog` is"
+                          " converted to float64. This behavior might change"
+                          " in the future.")
+
         # Get device array. Float64 only for now.
         self.d_y, self.n_obs, self.batch_size, self.dtype \
-            = input_to_cuml_array(endog, check_dtype=np.float64)
+            = input_to_cuml_array(endog, convert_to_dtype=np.float64)
```

Comment:
```suggestion
            = input_to_cuml_array(
                endog, 
                convert_to_dtype=(np.float64 if convert_dtype else None)
            )
```

We can add a `convert_dtype` parameter in the `__init__` (set by default to True) so that the user can disable the auto conversion if they want (similar to fit/predict in other models https://github.com/rapidsai/cuml/blob/af1317439c1aaf4235c28e1f202e2b8fb83d6391/python/cuml/linear_model/linear_regression.pyx#L222) what do you think?

Reply:
I have added `convert_dtype` and removed the warning (unless I make `False` the default for this option, it doesn't make sense to have a warning because this is the intended behavior).

I guess we can implement a warning when we add float32 support.